### DISCUSSION TITLE
Regarding pc_curve

### DISCUSSION THREAD

**Discussion by jayaraj777** [PMEAL/OpenPNM#2727] [Q&A]
I am trying to solve the invasion of air in a network initially filled with water. I used the IvasionPercolation algorithm for this. I was able to run the code but when I try to plot the capillary pressure vs air saturation, I am not getting a proper plot. For the same value of capillary pressure, I am getting many air saturation values. Please suggest me a solution for this.  
![image](https://user-images.githubusercontent.com/123013599/230710295-9786d1f0-8270-4d8c-a58a-0684bb2068f7.png)

Thanks
Jayaraj

**Comment by jgostick** [comment]
This curves is perfect!  Invasion percolation is the process of invading every single pore in sequence, so usually the pore/throat being invaded is large (low pressure), but then occasionally the fluid will breach a small pore/throat (high pressure) and the capillary pressure will spike.  The envelope of peak or spike pressures on the right-hand edge of the above data is the drainage capillary curve that you're looking for.  You can get the drainage curve using the 'Drainage' algorithm.

**Comment by jayaraj777** [comment]
Thanks for the reply sir. How can I get capillary pressure vs average air
saturation of the network curve. Please suggest me a solution.

Thanks
Jayaraj

**Reply by jgostick** [reply]
The number of tech support questions we receive lately is just too much for us to handle.  Our software is a tool, not a service, so we really have to limit how much time we spend on helping.  I'm sorry about that.  Here is [a link](https://openpnm.org/examples/tutorials/09_simulating_invasion.html) that explains the IP vs OP stuff that might be useful for you.

**Comment by jayaraj777** [comment]
I am very much grateful for your support. Sorry for the inconvenience
caused.

Thanks
Jayaraj
